src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -412,6 +457,44 @@ std::shared_ptr<Pane> Pane::NavigateDirection(const std::shared_ptr<Pane> source
         return PreviousPane(sourcePane);
     }
 
+    // Check if moving up or down the tree
+    if (direction == FocusDirection::Parent)
+    {
+        if (const auto parent = _FindParentOfPane(sourcePane))
+        {
+            // Capture a weak reference to the source pane if it is a leaf so that
+            // we can keep the same terminal focused and take the same path back down
+            // if a child movement is requested.
+            parent->_previouslyFocusedTerminal = sourcePane->_IsLeaf() ? sourcePane->weak_from_this() : sourcePane->_previouslyFocusedTerminal;

```

Comment:
do if you had `[1 | [ 2 | [3 | 4] ] ]` and went from 3->parent->parent->child, what's selected now? `3` or `[ 2 | [3 | 4] ]`?


This makes me think it's `3`, the leaf

Reply:
The result will be `[2 | [3 | 4] ]` since child movement just goes down one level.

Reply:
I tried to update the comment to make it more clear that we are explicitly storing a reference to the leaf and its TermControl.